diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/search/VariablesWithAssignmentSearch.kt

```diff
@@ -0,0 +1,39 @@
+package org.cqfn.diktat.ruleset.utils.search
+
+import com.pinterest.ktlint.core.ast.ElementType
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithSpecificType
+import org.cqfn.diktat.ruleset.utils.isGoingAfter
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtProperty
+
+class VariablesWithAssignmentSearch(fileNode: ASTNode,
+                                    filterForVariables: (KtProperty) -> Boolean) : VariableSearch(fileNode, filterForVariables) {
+
+    override fun KtElement.getAllSearchResults(property: KtProperty): List<KtNameReferenceExpression> {
+        return this.node.findAllNodesWithSpecificType(ElementType.BINARY_EXPRESSION)
+                // filtering out all usages that are declared in the same context but are going before the variable declaration
+                .filter {
+                    // FixMe: bug is here with a search of postfix/prefix variables assignment
```

Comment:
What do you mean? Operation like `++`, `+=`?

Reply:
yes, they will not work properly now, I will handle it